.github/workflows/check.yml

```diff
@@ -15,52 +15,18 @@ jobs:
         strategy:
             fail-fast: false
             matrix:
-                node-version: [10, 12, 14, 15]
+                node-version: [ 10, 12, 14, 15 ]
 
         steps:
-            -
-                uses: actions/checkout@v2
-            -
-                name: Use Node.js ${{ matrix.node-version }}
+            -   uses: actions/checkout@v2
+            -   name: Use Node.js ${{ matrix.node-version }}
                 uses: actions/setup-node@v1
                 with:
                     node-version: ${{ matrix.node-version }}
-            -
-                name: Cache Node Modules
-                if: ${{ matrix.node-version == 14 }}
-                uses: actions/cache@v2
-                with:
-                    path: |
-                        node_modules
-                        build
-                    key: cache-${{ github.run_id }}-v14
-            -
-                name: Install Dependencies
+            -   name: Install Dependencies
                 run: npm install
-            -
-                name: Run Tests
-                run: npm test
-
-    lint:
-        name: Lint
-        needs: [build_and_test]
-        runs-on: ubuntu-latest
-
-        steps:
-            -
-                uses: actions/checkout@v2
-            -
-                name: Use Node.js 14
-                uses: actions/setup-node@v1
-                with:
-                    node-version: 14
-            -
-                name: Load Cache
-                uses: actions/cache@v2
-                with:
-                    path: |
-                        node_modules
-                        build
-                    key: cache-${{ github.run_id }}-v14
-            -
+            -   name: Run Linter
+                if: matrix.node-version == '15'
```

Comment:
I don't like this very much, because failing lint will look like a failing test. I really like the separate block for linting in the test results, which is now gone. But I guess I can live with it. It simplifies the action quite a bit. Would be nice to find some other solution if possible.

![image](https://user-images.githubusercontent.com/23726914/112861453-17efbd80-90b5-11eb-9e4b-ba6f011fd97b.png)

Reply:
as said on slack, if we add lock file, we can cache the node modules as a whole (for all steps and all node versions), speed up all builds that do not change dependencies and keep it in separate step as it was before. the caching was the issue, and it was there only to speed up the lint step.

i am all in for the lock file, on the other hand i would vote for using yarn after seeing all the weirdness with npm's one :] but one way or the other, we need a lock file to properly invalidate the cache.

(btw I know that the cache key was based on the run id or something, but it was really causing failures for me, i guess the run id was reused when re-running failed builds which caused it)

Reply:
Or, heretic idea, we could keep the lint in a separate job, but install only `eslint` and `@apify/eslint-config` in the lint step. (and keep the cache removed)

Reply:
Yes, heretic indeed. I am offering the right way to allow node modules caching, which is the lock file. And it will have much better overall perf with that solution as you save the install time for each node version, for each build that is not fiddling with dependencies (most of them). 

Btw last time we argued about this, you said you are not aware of projects that would do it that way. So here few examples of the big guys, to get you outside of that weird apify bubble :] Note many of the big guys actually use yarn - it is definitely not something to be affraid of, yarn is great and in my eyes light years ahead of npm (since it was born, npm is just trying to catch up, copying its features, usually in a wrong/incomplete way). 

- angular, uses yarn: https://github.com/angular/angular (they even have a [short readme for the lock file](https://github.com/angular/angular/blob/master/yarn.lock.readme.md) with some reasoning about it)
- lerna, uses npm: https://github.com/lerna/lerna
- react, uses yarn: https://github.com/facebook/react
- vue, uses yarn: https://github.com/vuejs/vue
- ghost cms, uses yarn: https://github.com/TryGhost/Ghost
- bootstrap, uses npm: https://github.com/twbs/bootstrap
- typeorm, uses npm: https://github.com/typeorm/typeorm
- prisma, uses pnpm: https://github.com/prisma/prisma
- cheerio, uses npm: https://github.com/cheeriojs/cheerio
- docusaurus, uses yarn: https://github.com/facebook/docusaurus

So again, I am strongly in favour of going for yarn and lock files. From this quick search it actually looks like yarn is quite common in the big repositories.